apps/mobile/src/features/activity/activity-widget.tsx

```diff
@@ -1,30 +1,39 @@
 import { ScrollView } from 'react-native-gesture-handler';
 
-import { FetchState, FetchWrapper } from '@/components/loading';
+import { FetchState } from '@/components/loading';
 import { Widget } from '@/components/widget';
+import { ActivityCard } from '@/features/activity/activity-card';
 import { useTheme } from '@shopify/restyle';
 
-import { Activity } from '@leather.io/models';
+import { Activity, OnChainActivity, OnChainActivityTypes } from '@leather.io/models';
 import { Box, Theme } from '@leather.io/ui/native';
 
-import { ActivityEmpty } from './activity-empty';
-import { ActivityList } from './activity-list';
+const renderLimit = 10;
 
-export function hasActivity(activity: FetchState<Activity[]>) {
+function hasActivity(activity: FetchState<Activity[]>) {
   return activity.state === 'success' && activity.value.length > 0;
 }
 
+function inOnChainActivity(activity: Activity): activity is OnChainActivity {
+  return activity.type in OnChainActivityTypes;
+}
+
+function getVisibleActivity(activity: Activity[]) {
+  return activity.slice(0, renderLimit).filter(inOnChainActivity);
+}
+
 interface ActivityWidgetProps {
   activity: FetchState<Activity[]>;
   onPressHeader?: () => void;
   title: string;
 }
 export function ActivityWidget({ activity, onPressHeader, title }: ActivityWidgetProps) {
   const theme = useTheme<Theme>();
-  // Don't render anything if loading or no activity
-  if (activity.state === 'loading' || !hasActivity(activity)) {
+
+  if (activity.state !== 'success' || !hasActivity(activity)) {
```

Comment:
I don't think this will work for the empty case. It needs to be a success to know how many entries there are

Reply:
`hasActivity` was already checking for `success` so this should still work. This simply excludes the error state, since it was rendering a full page broken error boundary which I don't think was intended. 
On the long term I think those widgets need better visual handling of errors/loading/empty states. Logged an [issue](https://linear.app/leather-io/issue/LEA-2970/home-widgets-non-success-states).

Reply:
OK great 👍